android/src/main/java/com/reactnativearviewer/ArViewerView.kt

```diff
@@ -54,59 +72,346 @@ open class ArViewerView @JvmOverloads constructor(
+      val config = Config(session)
+
+      // Set plane orientation mode
+      updatePlaneDetection(config)
+      // Enable or not instant placement
+      updateInstantPlacement(config)
+      // Enable or not light estimation
+      updateLightEstimation(config)
+      // Enable or not depth management
+      updateDepthManagement(config)
+
+      config.focusMode = Config.FocusMode.AUTO
+      // Force the non-blocking mode for the session.
+      config.updateMode = Config.UpdateMode.LATEST_CAMERA_IMAGE
+      sessionConfig = config
+      arSession = session
+
+      arView!!.session?.configure(sessionConfig)
+      arView!!.session = arSession
+
+      initializeSession()
+      resume()
+
+      // Setup the instructions view.
+      instructionsController = InstructionsController(context, this);
+      instructionsController!!.setEnabled(true);
+    }
+  }
+
+  private fun resume() {
+    if (isStarted) {
+      return
+    }
+    if ((context as ThemedReactContext).currentActivity != null) {
+      isStarted = true
+      try {
+        arView!!.resume()
+      } catch (ex: java.lang.Exception) {
+        sessionInitializationFailed = true
+      }
+      if (!sessionInitializationFailed) {
+        instructionsController?.setVisible(true)
+      }
+    }
+  }
+
+  /**
+   * Initializes the ARCore session. The CAMERA permission is checked before checking the
+   * installation state of ARCore. Once the permissions and installation are OK, the method
+   * #getSessionConfiguration(Session session) is called to get the session configuration to use.
+   * Sceneform requires that the ARCore session be updated using LATEST_CAMERA_IMAGE to avoid
+   * blocking while drawing. This mode is set on the configuration object returned from the
+   * subclass.
+   */
+  private fun initializeSession() {
+    // Only try once
+    if (sessionInitializationFailed) {
+      return
+    }
+    // if we have the camera permission, create the session
+    if (CameraPermissionHelper.hasCameraPermission((context as ThemedReactContext).currentActivity)) {
+      val sessionException: UnavailableException?
+      try {
+        if (requestInstall()) {
+          return
+        }
+        onSessionConfigurationListener?.onSessionConfiguration(arSession, sessionConfig)
+
+        // run a JS event
+        Log.d("ARview session", "started")
+        val event = Arguments.createMap()
+        val reactContext = context as ThemedReactContext
+        reactContext.getJSModule(RCTEventEmitter::class.java).receiveEvent(
+          id,
+          "onStarted",
+          event
+        )
+
+        return
+      } catch (e: UnavailableException) {
+        sessionException = e
+      } catch (e: java.lang.Exception) {
+        sessionException = UnavailableException()
+        sessionException.initCause(e)
+      }
+      sessionInitializationFailed = true
+      returnErrorEvent(sessionException?.message)
+    } else {
+      requestDangerousPermissions()
```

Comment:
Aren't permissions managed on JS side? Is that still usefull ?

Reply:
Removed requestDangerousPermissions() but is CameraPermissionHelper.hasCameraPermission() still needed ?